### PR TITLE
Fixes for carter tracy aquifer model

### PR DESCRIPTION
**Pull request by kel85uk**
Changes:

* Makes the necessary interface changes to check for nullptr instead of values for the influx_coeff and aquifer initial pressure.

* Corrects for the issue where the aquifer makes connections to cell faces adjoining active cells (see AQUANCON item 11).

Upstream: https://github.com/OPM/opm-common/pull/559

### CODE REVIEW

**Comment by kel85uk** [comment]
Has to be accepted once this PR has been accepted in the corresponding OPM-Common module https://github.com/OPM/opm-common/pull/476

**Comment by kel85uk** [comment]
I think the rebase has messed up some files. Do correct me if am wrong?

**Comment by joakim-hove** [comment]
The upstream PR: https://github.com/OPM/opm-common/pull/559 is ready to be merged; is this ready?

File: opm/autodiff/AquiferCarterTracy.hpp [hunk 1]
```diff
@@ -300,7 +300,18 @@ namespace Opm
                             // Check now if the face is outside of the reservoir, or if it adjoins an inactive cell
                             // Do not make the connection if the product of the two cellIdx > 0. This is because the
                             // face is within the reservoir/not connected to boundary. (We still have yet to check for inactive cell adjoining)
-                            faceArea = (faceCells(faceIdx,0)*faceCells(faceIdx,1) > 0)? 0. : Opm::UgGridHelpers::faceArea(ugrid, faceIdx);
+                            auto cellNeighbour0 = faceCells(faceIdx,0);
```
Comment (GitPaean) [inline opm/autodiff/AquiferCarterTracy.hpp]: This part of the code can be put into a function to calculate face area, so the following code can become

```
faceArea_connected_.at(idx) = faceArea(.....);
```

File: opm/autodiff/AquiferCarterTracy.hpp [hunk 2]
```diff
@@ -300,7 +300,18 @@ namespace Opm
...
+                            auto cellNeighbour1 = faceCells(faceIdx,1);
+                            auto defaultFaceArea = Opm::UgGridHelpers::faceArea(ugrid, faceIdx);
```
Comment (GitPaean) [inline opm/autodiff/AquiferCarterTracy.hpp]: const when possible please

**Comment by atgeirr** [comment]
jenkins build this opm-common=559 please

**Comment by hnil** [comment]
Tested this on a large model. Error seemed to be that the face tag was called at cell not active in the grid. Will check this closer.

**Comment by hnil** [comment]
If I am correct will this test which obvoisly is wrong int put git segmentation fault.
'''
AQUANCON
-- Aq#     I1 I2  J1   J2  K1 K2 FACE
    1      1  1  4    6   1  1  J- 1.0 1.0 NO/
/
,,,

**Comment by hnil** [comment]
Minimal failing example:  I am not sure, but it seems like cell indexes is not mapped from logical cartesian to active cell numbering. Setting aquifer in last logical cartesian cell and actnum way one will trigger the error: 

ACTNUM
4*1
1*0
5*1
5*1
5*1
20*1
20*1
20*1
20*1
/
AQUANCON
-- Aq#     I1 I2  J1   J2  K1 K2 FACE
    1      20  20  1    1   5  5  J- 1.0 1.0 NO/
/

**Comment by hnil** [comment]
Mapping from logical grid can be done as:
void WellsManager::setupCompressedToCartesian i WellsManager.cpp

**Comment by kel85uk** [comment]
@hnil. I think it might be better to report the behaviour that you've mentioned in a separate issue and not in this PR? What do you think @GitPaean, @bska ?

**Comment by GitPaean** [comment]
@kel85uk , we do not have access to the correctness of the results here. 

Does this PR fix the problem existed in your side?

**Comment by GitPaean** [comment]
jenkins build this opm-common=559 please

**Comment by GitPaean** [comment]
@joakim-hove , @kel85uk and @nairr confirmed that this PR and OPM/opm-common#559 fixes their problem. I think both of these PRs are ready to go. Feel free to press green.

**Comment by GitPaean** [comment]
jenkins build this opm-common=559 please

**Pull request merged**